cibyl/sources/git.py

```diff
@@ -0,0 +1,91 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+import logging
+import os
+from functools import partial
+
+from git import Repo
+
+from cibyl.exceptions.git import GitError
+from cibyl.sources.source import Source, safe_request_generic
+
+LOG = logging.getLogger(__name__)
+
+safe_request = partial(safe_request_generic,
+                       custom_error=GitError)
+
+
+class GitSource(Source):
+    """A class representation of a Git-based source."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, url: str = None, dest: str = None, branch: str = None,
+                 enabled: bool = True, priority: int = 0,
+                 name: str = "jenkins_job_builder",
```

Comment:
I would change the default value for name and driver to something generic, or an empty string